java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1434 @@
+      return 0;
+    }
+    int totalLength = 0;
+    for (int i = 0; i < valueCount; i++) {
+      totalLength += getLength(i);
+    }
+    return totalLength;
+  }
+
+  /**
+   * Get the size (number of bytes) of underlying buffers used by this
+   * vector.
+   * @return size of underlying buffers.
+   */
+  @Override
+  public int getBufferSize() {
+    return getBufferSizeFor(this.valueCount);
+  }
+
+  /**
+   * Get the potential buffer size for a particular number of records.
+   * @param valueCount desired number of elements in the vector
+   * @return estimated size of underlying buffers if the vector holds
+   *         a given number of elements
+   */
+  @Override
+  public int getBufferSizeFor(final int valueCount) {
+    if (valueCount == 0) {
+      return 0;
+    }
+
+    final int validityBufferSize = getValidityBufferSizeFromCount(valueCount);
+    final int viewBufferSize = valueCount * ELEMENT_SIZE;
+    final int dataBufferSize = getDataBufferSize();
+    return validityBufferSize + viewBufferSize + dataBufferSize;
+  }
+
+  private int getDataBufferSize() {
+    int dataBufferSize = 0;
+    for (ArrowBuf buf : dataBuffers) {
+      dataBufferSize += (int) buf.writerIndex();
+    }
+    return dataBufferSize;
+  }
+
+  /**
+   * Get information about how this field is materialized.
+   * @return the field corresponding to this vector
+   */
+  @Override
+  public Field getField() {
+    return field;
+  }
+
+  /**
+   * Return the underlying buffers associated with this vector. Note that this doesn't
+   * impact the reference counts for this buffer, so it only should be used for in-context
+   * access. Also note that this buffer changes regularly, thus
+   * external classes shouldn't hold a reference to it (unless they change it).
+   * <p>
+   * Note: This method only returns validityBuffer and valueBuffer.
+   * But it doesn't return the data buffers.
+   * <p>
+   * TODO: Implement a strategy to retrieve the data buffers.
+   * <a href="https://github.com/apache/arrow/issues/40930">data buffer retrieval.</a>
+   *
+   * @param clear Whether to clear vector before returning, the buffers will still be refcounted
+   *              but the returned array will be the only reference to them
+   * @return The underlying {@link ArrowBuf buffers} that is used by this
+   *         vector instance.
+   */
+  @Override
+  public ArrowBuf[] getBuffers(boolean clear) {
+    final ArrowBuf[] buffers;
+    setReaderAndWriterIndex();
+    if (getBufferSize() == 0) {
+      buffers = new ArrowBuf[0];
+    } else {
+      buffers = new ArrowBuf[2];
+      buffers[0] = validityBuffer;
+      buffers[1] = viewBuffer;
+    }
+    if (clear) {
+      for (final ArrowBuf buffer : buffers) {
+        buffer.getReferenceManager().retain();
+      }
+      clear();
+    }
+    return buffers;
+  }
+
+  /**
+   * Validate the scalar values held by this vector.
+   */
+  public void validateScalars() {
+    // No validation by default.
```

Comment:
Is there an issue filed for this? We should be able to validate the length and so on.

Reply:
Not specifically, this was the same case for the `VarCharVector` representation.

Reply:
I left it as it is for the `VarCharVector`, and this was a [recently added feature](https://github.com/vibhatha/arrow/commit/a0041027902ac71f00572d5b31022ee98502ed6a).